### PR TITLE
dev-python/python-axolotl(-curve25519): New packages

### PR DESCRIPTION
**Pull request by ConiKost**
dev-python/python-axolotl(-curve25519): New packages

Closes: https://bugs.gentoo.org/608882

### CODE REVIEW

File: dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+MY_P="python-${P}"
```
Comment (mgorny) [inline dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1.ebuild]: This looks backwards. Define `MY_P` as `MY_PN-PV`.

File: dev-python/axolotl/axolotl-0.1.42.ebuild [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="dev-python/axolotl-curve25519[${PYTHON_USEDEP}]
+	dev-python/protobuf-python[${PYTHON_USEDEP}]
+	dev-python/pycrypto[${PYTHON_USEDEP}]
+	dev-python/setuptools[${PYTHON_USEDEP}]"
+
+DEPEND="${DEPEND}"
+
+S="${WORKDIR}/${MY_P}"
```
Comment (mgorny) [inline dev-python/axolotl/axolotl-0.1.42.ebuild]: Tests!

File: dev-python/axolotl/axolotl-0.1.42.ebuild [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+SRC_URI="https://github.com/tgalal/${MY_PN}/archive/${PV}.tar.gz -> ${MY_P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+RDEPEND="dev-python/axolotl-curve25519[${PYTHON_USEDEP}]
+	dev-python/protobuf-python[${PYTHON_USEDEP}]
+	dev-python/pycrypto[${PYTHON_USEDEP}]
+	dev-python/setuptools[${PYTHON_USEDEP}]"
```
Comment (mgorny) [inline dev-python/axolotl/axolotl-0.1.42.ebuild]: Are setuptools needed at runtime?

File: dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1.ebuild (hunk 1 above)
Comment (ConiKost) [reply dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1.ebuild]: Ok, I changed it, to make it better, in both eBuilds.

File: dev-python/axolotl/axolotl-0.1.42.ebuild (hunk 2 above)
Comment (ConiKost) [reply dev-python/axolotl/axolotl-0.1.42.ebuild]: Added and they pass locally for me fine.

File: dev-python/axolotl/axolotl-0.1.42.ebuild (hunk 3 above)
Comment (ConiKost) [reply dev-python/axolotl/axolotl-0.1.42.ebuild]: No, moved to DEPEND.

File: dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+MY_PN="python-${PN}"
+MY_PV="${PV}-2"
```
Comment (mgorny) [inline dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1.ebuild]: Include that -2 into PV, and transform it back (e.g. `0.4.1_p2` would be an easy way to do it).

File: dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+PYTHON_COMPAT=( python2_7 python3_{4,5,6} )
+
+inherit distutils-r1 eapi7-ver
+
+DESCRIPTION="A python wrapper for the curve25519 library with ed25519 signatures"
+HOMEPAGE="https://github.com/tgalal/python-axolotl-curve25519"
+SRC_URI="https://github.com/tgalal/${MY_PN}/archive/${MY_PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
```
Comment (mgorny) [inline dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1.ebuild]: Missing license: https://github.com/tgalal/python-axolotl-curve25519/blob/master/curve/curve25519-donna.c#L1

Comment (ConiKost) [reply dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1.ebuild]: `BSD` added.

File: dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1.ebuild (hunk 4 above)
Comment (ConiKost) [reply dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1.ebuild]: Thanks, changed.

**Comment by mgorny** [comment]
Reverted the mistaken merge.

File: dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1_p2.ebuild [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit eapi7-ver
+
+MY_PN="python-${PN}"
```
Comment (mgorny) [inline dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1_p2.ebuild]: Hmm, actually, why aren't you using the upstream name? We have a number of packages starting with `dev-python/python-`, and this package is also named `python-axolotl` on pypi, so I think that'd be a more correct name here.

File: dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1_p2.ebuild [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+MY_PV="$(ver_cut 1-3)-$(ver_cut 5)"
```
Comment (mgorny) [inline dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1_p2.ebuild]: Simpler, and more indicative of what you're actually doing to PV:

    ${PV/_p/-}

File: dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1_p2.ebuild [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+PYTHON_COMPAT=( python2_7 python3_{4,5,6} )
+
+inherit distutils-r1 eapi7-ver
+
+DESCRIPTION="A python wrapper for the curve25519 library with ed25519 signatures"
+HOMEPAGE="https://github.com/tgalal/python-axolotl-curve25519"
+SRC_URI="https://github.com/tgalal/${MY_PN}/archive/${MY_PV}.tar.gz -> ${P}.tar.gz"
```
Comment (mgorny) [inline dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1_p2.ebuild]: Use MY_PN-MY_PV as distfile name to match directory from S.

File: dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1_p2.ebuild (hunk 6 above)
Comment (ConiKost) [reply dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1_p2.ebuild]: Okay, changed to python-*.

File: dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1_p2.ebuild (hunk 7 above)
Comment (ConiKost) [reply dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1_p2.ebuild]: Done.

File: dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1_p2.ebuild (hunk 8 above)
Comment (ConiKost) [reply dev-python/axolotl-curve25519/axolotl-curve25519-0.4.1_p2.ebuild]: Done.

File: dev-python/python-axolotl/python-axolotl-0.1.42.ebuild [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 python3_{4,5,6} )
+
+inherit distutils-r1
+
+DESCRIPTION="A python module for the axolotl protocol"
+HOMEPAGE="https://github.com/tgalal/python-axolotl"
+SRC_URI="https://github.com/tgalal/${MY_PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
```
Comment (mgorny) [inline dev-python/python-axolotl/python-axolotl-0.1.42.ebuild]:
```
>>> Emerging (4 of 4) dev-python/python-axolotl-0.1.42::gentoo
>>> Downloading 'https://github.com/tgalal//archive/0.1.42.tar.gz'
--2018-06-23 18:10:46--  https://github.com/tgalal//archive/0.1.42.tar.gz
Resolving github.com (github.com)... 192.30.253.112, 192.30.253.113
Connecting to github.com (github.com)|192.30.253.112|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2018-06-23 18:10:47 ERROR 404: Not Found.

Comment (ConiKost) [reply dev-python/python-axolotl/python-axolotl-0.1.42.ebuild]: *grml*, sorry, should be now fixed.
Is there some debug switch, to force always a new download for SRC_URI during testing and ignoring local download?

Comment (mgorny) [reply dev-python/python-axolotl/python-axolotl-0.1.42.ebuild]: I use `export DISTDIR=/tmp/dist` while testing ;-).

Comment (mgorny) [reply dev-python/python-axolotl/python-axolotl-0.1.42.ebuild]:
```
!!! Fetched file: python-axolotl-0.1.42.tar.gz VERIFY FAILED!
!!! Reason: Filesize does not match recorded size
!!! Got:      53869
!!! Expected: 39472

Comment (ConiKost) [reply dev-python/python-axolotl/python-axolotl-0.1.42.ebuild]: * python-axolotl-0.1.42.tar.gz BLAKE2B SHA512 size ;-) ...                                                                                                                                                                        [ ok ]
>>> Unpacking source...
>>> Unpacking python-axolotl-0.1.42.tar.gz to /var/tmp/portage/dev-python/python-axolotl-0.1.42/work
>>> Source unpacked in /var/tmp/portage/dev-python/python-axolotl-0.1.42/work
>>> Preparing source in /var/tmp/portage/dev-python/python-axolotl-0.1.42/work/python-axolotl-0.1.42 ...

Emerging works here fine? :/ Using the Manifest from my commit here.

Comment (ConiKost) [reply dev-python/python-axolotl/python-axolotl-0.1.42.ebuild]: Ah. I see the problem. The `python-axolotl-0.1.42.tar.gz` on the gentoo mirrors is faulty.. It differs from the one on Github upstream. Pushed correct manifest.

File: dev-python/python-axolotl-curve25519/python-axolotl-curve25519-0.4.1_p2.ebuild [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+DEPEND="dev-python/setuptools[${PYTHON_USEDEP}]"
+
+S="${WORKDIR}/${PN}-${MY_PV}"
+
+python_test() {
+	esetup.py test
```
Comment (mgorny) [inline dev-python/python-axolotl-curve25519/python-axolotl-curve25519-0.4.1_p2.ebuild]:
```
Ran 0 tests in 0.000s

OK

Comment (ConiKost) [reply dev-python/python-axolotl-curve25519/python-axolotl-curve25519-0.4.1_p2.ebuild]: I tend to drop the tests for `python-axolotl-curve25519`. Yes, I saw that thing with 0 tests, but let it for the case in. There are 2 tests under `python-axolotl-curve25519-0.4.1-2/curve/ed25519/tests`, but it seems, they are incomplete, as they can't run compile standalone, missing the main. When looking into `libsignal-protocol-c`, from this got ported, this seems only be a small port of the test suite, which can't be alone this way. At least, I think so, I am not an expert here.

**Comment by mgorny** [comment]
Thanks!

**Pull request closed without merging**